internal/commands/app/diff.go

```diff
@@ -125,12 +144,34 @@ func (cmd *CommandDiff) Handler(profile *user.Profile, ui terminal.UI, clients c
 }
 
 func (i *diffInputs) Resolve(profile *user.Profile, ui terminal.UI) error {
-	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory, true); err != nil {
+	searchPath := i.LocalPath
+	if searchPath == "" {
+		searchPath = profile.WorkingDirectory
+	}
+
+	app, err := local.LoadAppConfig(searchPath)
+	if err != nil {
 		return err
 	}
 
 	if i.LocalPath == "" {
-		i.LocalPath = profile.WorkingDirectory
+		if app.RootDir != "" {
+			i.LocalPath = app.RootDir
+		} else {
+			if err := ui.AskOne(&i.LocalPath, &survey.Input{Message: "App filepath (local)"}); err != nil {
```

Comment:
should we be checking if this `i.LocalPath != ""` or will the AskOne take care of that?

Reply:
the larger if block around this is `i.LocalPath == ""`, so we're only setting `LocalPath` if it wasn't set

Reply:
I meant afterwards when we try to use it to get the app. Is the error user friendly when the user inputs an empty string here on the prompt?

Reply:
yeah we return the error in the handler:

```go
	if app.RootDir == "" {
		return fmt.Errorf("no app directory found at %s", cmd.inputs.LocalPath)
	}
```